x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/threshold_edit/threshold_edit.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { type FieldSpec } from '@kbn/data-views-plugin/common';
+import { type FieldHook, UseMultiFields } from '../../../../shared_imports';
+import { ThresholdInput } from '../../../rule_creation_ui/components/threshold_input';
+import {
+  THRESHOLD_FIELD_CONFIG,
+  THRESHOLD_VALUE_CONFIG,
+  getCardinalityFieldConfig,
+  getCardinalityValueConfig,
+} from './field_configs';
+
+interface ThresholdEditProps {
+  path: string;
+  esFields: FieldSpec[];
+}
+
+export function ThresholdEdit({ path, esFields }: ThresholdEditProps): JSX.Element {
+  const ThresholdInputChildren = useCallback(
+    ({
+      thresholdField,
+      thresholdValue,
+      thresholdCardinalityField,
+      thresholdCardinalityValue,
+    }: Record<string, FieldHook>) => (
+      <ThresholdInput
+        browserFields={esFields}
```

Comment:
Repeating my earlier question: should we maybe just `filter` `esFields` within this component?